query-engine/schema/src/query_schema.rs

```diff
@@ -96,10 +107,31 @@ impl QuerySchema {
                 || self.has_capability(ConnectorCapability::FullTextSearchWithIndex))
     }
 
+    /// Returns whether the loaded connector supports the join strategy.
     pub fn can_resolve_relation_with_joins(&self) -> bool {
-        self.has_feature(PreviewFeature::RelationJoins)
-            && (self.has_capability(ConnectorCapability::LateralJoin)
-                || self.has_capability(ConnectorCapability::CorrelatedSubqueries))
+        !matches!(self.join_strategy_support, JoinStrategySupport::No)
```

Comment:
Shouldn't this also match `JoinStraregySupport::UnsupportedDbVersion`?

Reply:
no because it's used for query validation and the like, and not in `get_relation_load_strategy`

Reply:
No, because we use this everywhere except when computing the RelationLoadStrategy for a given query.

We want this to return true when `UnsupportedDbVersion` is selected, because we need the Query Schema to render the relation load args, or else we’d get desync with the client